src/redux/users/action.ts

```diff
@@ -0,0 +1,48 @@
+import { SignInData } from '../../components/_templates/SignIn/api'
+import {
+  error,
+  finish,
+  loading,
+  setUsers,
+  usersLoadSuccess,
+} from '../reducers/usersSlice'
+import { AppThunk } from '../store'
+import { reqestSignIn as requestSignInAPI } from '../../components/_templates/SignIn/api'
+import { reqestUserInfo as reqestUserInfoAPI } from '../../components/_templates/UserList/api'
+
+export type RequestSignInActionProps = {
+  users: SignInData
+  onSuccess?: Function
+  onError?: Function
+}
+
+export const requestSignIn =
+  ({ users, onSuccess, onError }: RequestSignInActionProps): AppThunk =>
+  async (dispatch) => {
+    try {
+      dispatch(loading())
+      const { data } = await requestSignInAPI(users)
+      if (data) {
+        dispatch(setUsers({ data: data.users }))
+        localStorage.setItem('token', data.token)
+        onSuccess && onSuccess()
```

Comment:
Can not say what I'm sure? but why u didn't pass anything to "onSuccess" function?